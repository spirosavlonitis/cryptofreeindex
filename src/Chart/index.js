import React from 'react';
import Form from 'react-bootstrap/lib/Form';
import FormControl from 'react-bootstrap/lib/FormControl';
import Radio from 'react-bootstrap/lib/Radio'

export default ({chartCoin, navCoins, onCoinChange, onDaysChange, activeKey, chartDays}) => {
	const navCoin = coin => coin !== navCoins[activeKey];
	const days = [
			[7, "1 Week"], [30,"1 Month"], [90, "3 Months"], [180, "6 Months"], 
			[365, "1 Year"],
	]

	return(
		<div>
			<div id="chartdiv" style={{ width: "100%", height: "500px" }}></div>
			<Form>
				{days.map(item => 
					<Radio
						key={`${item[0]}Radio`} 
						name="daysGroup" 
						value={item[0]}
						checked= {item[0] === chartDays ?  'checked' : '' }
						onChange={onDaysChange}
						inline 
					>
						{item[1]}
					</Radio>
					)
				}
				<FormControl 
					componentClass="select" 
					placeholder="select" 
					value={chartCoin || ''}  
					onChange={onCoinChange} 
				>
					{ navCoins.filter(navCoin).map(item => 
							<option key={`${item}Option`} value={item}  >{item}</option>
						)  
					}
				</FormControl>
			</Form>
		</div>
	)
}

