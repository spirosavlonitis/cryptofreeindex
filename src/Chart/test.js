import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Chart from './index.js'


describe('Chart', () => {

	const props = {
		navCoins: [ 
				'USD', 'EUR', 'GBP', 'BTC', 'LTC', 'BCH', 'ETH',
  			'ETC', 'ZEC', 'DASH', 'XMR', 'DCR' 
  	],
  	chartCoin: 'BTC',
  	activeKey: 0,
  	onChange: () => 'foo',
	};

	it('renders without crashing', () =>{
		const div = document.createElement('div');
		ReactDOM.render(<Chart {...props} />, div);
		ReactDOM.unmountComponentAtNode(div);
	})

	test('snapshot', ()=>{
		const element = renderer.create(<Chart {...props} />);
		let tree = element.toJSON();
		expect(tree).toMatchSnapshot();
	})

})