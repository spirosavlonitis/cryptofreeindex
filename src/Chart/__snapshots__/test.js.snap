// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart snapshot 1`] = `
<div>
  <div
    id="chartdiv"
    style={
      Object {
        "height": "500px",
        "width": "100%",
      }
    }
  />
  <form
    className=""
  >
    <select
      className="form-control"
      placeholder="select"
      value="BTC"
    >
      <option
        value="USD"
      >
        USD
      </option>
      <option
        value="EUR"
      >
        EUR
      </option>
      <option
        value="GBP"
      >
        GBP
      </option>
      <option
        value="BTC"
      >
        BTC
      </option>
      <option
        value="LTC"
      >
        LTC
      </option>
      <option
        value="BCH"
      >
        BCH
      </option>
      <option
        value="ETH"
      >
        ETH
      </option>
      <option
        value="ETC"
      >
        ETC
      </option>
      <option
        value="ZEC"
      >
        ZEC
      </option>
      <option
        value="DASH"
      >
        DASH
      </option>
      <option
        value="XMR"
      >
        XMR
      </option>
      <option
        value="DCR"
      >
        DCR
      </option>
    </select>
  </form>
</div>
`;
